Handle trailing slash in `createURLRoute`

I came across this issue today where `createURLRoute(import.meta.env.VITE_DEV_SERVER_URL, id)` generated an invalid url because `import.meta.env.VITE_DEV_SERVER_URL` had a trailing `/`. 

The generated URL was something like `http://localhost:1234//#/main`.

I was able to fix it via `url.replace(/\/+$/, '')` but I still think this should be handled by this package.

Yeah, all good just wanted to call out an edge case. I did run into a build issue with v1.0.3 though:

```
[ERROR] Could not read from file: /Users/ogolan/src/desktop-app/packages/renderer/src/shared

    node_modules/electron-router-dom/dist/esm/index.js:1:183:
      1 │ ...r-dom";import{removeURLExtraDoubleSlashes as c}from"src/shared";function i(o){return Object.keys(o).reduce((t,e)=>...
        ╵                                                       ~~~~~~~~~~~~

/Users/ogolan/src/desktop-app/node_modules/esbuild/lib/main.js:1566
  let error = new Error(`${text}${summary}`);
```

--- a/README.md
+++ b/README.md
@@ -238,6 +238,13 @@ Example:
 ### Route
 It's the [react-router-dom](https://reactrouter.com/en/main/route/route#type-declaration) `<Route />` component, same props, same usage. 😄
 
+### In-depth example
+The recommended way to go to handle window states between main and renderer process (like creating or closing a new window) is using [IPC](https://www.electronjs.org/docs/latest/tutorial/ipc). You can check the [electron-app](https://github.com/daltonmenezes/electron-app) boilerplate to see how it was achieved, like:
+
+- [A preload script requesting window creation using IPC](https://github.com/daltonmenezes/electron-app/blob/main/src/preload/ipcs/windows/about/create.ts)
+- [A window creation on main process using IPC](https://github.com/daltonmenezes/electron-app/blob/main/src/main/windows/About/ipcs/register-window-creation.ts)
+- [The Routing doc from electron-app boilerplate](https://github.com/daltonmenezes/electron-app/blob/main/docs/ROUTING.md)
+
 # Contributing
 > **Note**: contributions are always welcome, but always **ask first**, — please — before work on a PR.
 

--- a/package.json
+++ b/package.json
@@ -1,6 +1,6 @@
 {
   "name": "electron-router-dom",
-  "version": "1.0.2",
+  "version": "1.0.3",
   "main": "dist/cjs/index.js",
   "module": "dist/esm/index.js",
   "types": "dist/index.d.ts",

--- a/package.json
+++ b/package.json
@@ -1,6 +1,6 @@
 {
   "name": "electron-router-dom",
-  "version": "1.0.3",
+  "version": "1.0.4",
   "main": "dist/cjs/index.js",
   "module": "dist/esm/index.js",
   "types": "dist/index.d.ts",

--- a/src/main/modules/routes/create-url.ts
+++ b/src/main/modules/routes/create-url.ts
@@ -1,4 +1,4 @@
-import { removeURLExtraDoubleSlashes } from 'src/shared'
+import { removeURLExtraDoubleSlashes } from '../../../shared'
 
 export function createURLRoute(route: string, id: string) {
   const URL = `${route}/#/${id}`

--- a/src/main/modules/routes/create-url.ts
+++ b/src/main/modules/routes/create-url.ts
@@ -1,3 +1,7 @@
+import { removeURLExtraDoubleSlashes } from 'src/shared'
+
 export function createURLRoute(route: string, id: string) {
-  return `${route}/#/${id}`
+  const URL = `${route}/#/${id}`
+
+  return removeURLExtraDoubleSlashes(URL)
 }

--- a/src/shared/index.ts
+++ b/src/shared/index.ts
@@ -1,1 +1,2 @@
+export * from './utils/remove-url-extra-double-slashes'
 export * from './utils/to-lower-case-keys'

--- a/src/shared/utils/remove-url-extra-double-slashes.ts
+++ b/src/shared/utils/remove-url-extra-double-slashes.ts
@@ -1,0 +1,3 @@
+export function removeURLExtraDoubleSlashes(url: string) {
+  return url.replace(/([^:]\/)\/+/g, '$1')
+}

